packages/validator/src/validator.ts

```diff
@@ -265,6 +294,40 @@ export class Validator {
       return BeaconHealth.ERROR;
     }
   }
+
+  private async fetchValidatorStatuses(pubkeysHex: string[], metrics: Metrics | null): Promise<void> {
+    const res = await this.api.beacon.getStateValidators("head", {
+      id: pubkeysHex,
+    });
+    ApiError.assert(res, "Can not fetch state validators from beacon node");
+
+    const allValidatorStatuses = new MapDef<SimpleValidatorStatus, number>(() => 0);
+
+    for (const validatorState of res.response.data) {
+      // Group all validators by status
+      const status = statusToSimpleStatusMapping(validatorState.status);
+      allValidatorStatuses.set(status, allValidatorStatuses.getOrDefault(status) + 1);
+    }
+
+    // The number of validators that are not in the beacon chain
+    const pendingCount = pubkeysHex.length - res.response.data.length;
+
+    allValidatorStatuses.set("pending", allValidatorStatuses.getOrDefault("pending") + pendingCount);
+
+    // Retrieve the number of validators for each status
+    const statuses = Object.fromEntries(Array.from(allValidatorStatuses.entries()).filter((entry) => entry[1] > 0));
+
+    // The total number of validators
+    const total = pubkeysHex.length;
+
+    if (metrics) {
+      for (const [status, amount] of allValidatorStatuses) {
+        metrics.validatorStatuses.set({status}, amount);
```

Comment:
shouldn't this rather be called `count`?

```suggestion
      for (const [status, count] of allValidatorStatuses) {
        metrics.validatorStatuses.set({status}, count);
```